File: libsim/src/HWIO.cpp
#include <ece643/libhwio/HWIO.hpp>

using namespace ece643::libhwio;

HWIO::HWIO() noexcept : mmap(0, 0), interrupt(mmap), led(mmap), ps2(mmap), vga(mmap) {
}

HWIO::HWIO(HWIO &&move) noexcept : mmap(0, 0), interrupt(mmap), led(mmap), ps2(mmap), vga(mmap) {
}

HWIO &HWIO::operator =(HWIO &&move) noexcept {
    return *this;
}

File: sim/src/main/kotlin/ece643/libhwio/HWIO.kt
package ece643.libhwio

import ece643.sim.SimWindow

class HWIO(window: SimWindow) {
    val accelerometer = Accelerometer(window.accelerometer)
    val led = LED(window.leds, window.sevenseg)
    val ps2 = PS2(window.ps2)
    val interrupt = Interrupt(window, window.buttons, ps2)
    val vga = VGA(window.vga)
}
